apps/omg_eth/lib/omg_eth/root_chain.ex

```diff
@@ -294,7 +294,7 @@ defmodule OMG.Eth.RootChain do
   # TODO - missing description + could this be moved to a statefull process?
   @spec get_root_deployment_height(binary() | nil, optional_addr_t()) ::
           {:ok, integer()} | Ethereumex.HttpClient.error()
-  def get_root_deployment_height(txhash \\ nil, contract \\ nil) do
+  def get_root_deployment_height(txhash \\ nil, contract \\ %{}) do
```

Comment:
if we're going down the route of replacing contracts with a map `%{}`, maybe a struct would be a better representation what's being expected?

Reply:
yeah, let's maybe tackle separately, this is the part that's not fully fleshed out yet (contract address management), but I think pieces of the puzzle are slowly falling into their places now.